hack/go-build.sh

```diff
@@ -5,5 +5,5 @@ set -e
 source hack/common.sh
 
 mkdir -p ${OUTDIR_BIN}
-go build -i -ldflags="-s -w" -o ${OUTDIR_BIN}/ocs-operator ./cmd/manager
```

Comment:
Why did you remove the "-i" flag?

Reply:
The -i flag installs the packages that are dependencies of the target.

Which we have already done IMO and it was causing the below error which I did not found any solution.

```
$ go build -tags 'netgo osusergo' -i -ldflags="-s -w" -o _output/tools/bin/ocs-operator ./cmd/manager
go build os/user: copying /home/nigoyal/.cache/go-build/47/47f22a329c0d011f534ff48e6501350f57139e9dd43c056e0856da7c8d4739e4-d: open /usr/lib/golang/pkg/linux_amd64/os/user.a: permission denied
go build net: copying /home/nigoyal/.cache/go-build/dd/dd6a332ebdbafca3686494fa00f07973256745f3bbd9da76887318b33e5ed44e-d: open /usr/lib/golang/pkg/linux_amd64/net.a: permission denied
```
The above was not working out even after giving sudo and moreover, I don't want to give sudo permissions at all just for building one go program.
```
$ go build -tags 'netgo osusergo' -ldflags="-s -w" -o _output/tools/bin/ocs-operator ./cmd/manager
Success
```

Reply:
With that explanation, the change is acceptable to me. 

Please include this in the commit message though.

Reply:
will do

Reply:
Honestly I am not very comfortable accepting `it was causing the below error which I did not found any solution` as a reason at the moment. At least we should know why we hit that issue? Could it be solved in some other way? Are we sure removing this flag won't backfire?

All these questions because I'm not well versed with Go builds.

Reply:
Agreed with Umanga, we should do a bit more research. Apart from a more detailed explanation, it should be its own commit.